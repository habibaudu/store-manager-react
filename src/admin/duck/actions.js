import types from './types';

const setAllProducts = products => ({
  type: types.LOAD_ALL_PRODUCT,
  payload: products,
});

const addProducts = productadded => ({
  type: types.ADD_PRODUCT,
  productadded,
});

const addProductStatus = status => ({
  type: types.SET_ADD_STATE,
  status,
});

const addProductError = errormessage => ({
  type: types.SET_ADD_ERROR,
  errormessage,
});

const setDeleteProduct = id => ({
  type: types.DELETE_PRODUCT,
  id,
});

const setSignupState = signupState => ({
  type: types.SET_SIGNUP_STATE,
  signupState,
});
const setSignupError = errorMessage => ({
  type: types.SET_SIGNUP_ERROR,
  errorMessage,
});

export default {
  setAllProducts,
  addProducts,
  addProductStatus,
  addProductError,
  setDeleteProduct,
  setSignupState,
  setSignupError,
};
