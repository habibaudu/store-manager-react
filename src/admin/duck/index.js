export { default as adminReducer } from './reducers';
export { default as constants } from './constants';
export { fetchAllProducts, deleteProduct, createAttendant } from './operations';
export { addProduct } from './operations';
export { default as actions } from './actions';
export { default as types } from './types';
