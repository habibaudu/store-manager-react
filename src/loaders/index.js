export { default as EllipsisLoaderComponent } from './EllipsisLoaderComponent';
export { default as PreloaderComponent } from './PreloaderComponent';

