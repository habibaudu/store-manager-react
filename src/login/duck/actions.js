import types from './types';

const setLoginState = (loginState, loginData) => ({
  type: types.SET_LOGIN_STATE,
  loginState,
  loginData,
});

const setLoginError = errorMessage => ({
  type: types.SET_LOGIN_ERROR,
  errorMessage,
});

export default { setLoginState, setLoginError };
