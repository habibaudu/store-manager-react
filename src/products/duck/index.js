export { default as productReducer } from './reducers';
export { default as constants } from './constants';
export { fetchAllProducts, searchProduct } from './operations';
export { default as actions } from './actions';
export { default as types } from './types';
