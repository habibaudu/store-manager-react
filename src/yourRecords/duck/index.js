export { default as recordReducer } from './reducers';
export { default as constants } from './constants';
export { fetchMyRecords } from './operations';
export { default as actions } from './actions';
export { default as types } from './types';
