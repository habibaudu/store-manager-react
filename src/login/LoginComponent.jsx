import React, { Fragment } from 'react';
import { Redirect, Link } from 'react-router-dom';
import { constants } from './duck';
import jwtDecode from 'jwt-decode';

export const LoginComponent = ({ loginUser, loginState }) => {
  const onFormSubmit = e => {
    e.preventDefault();
    const email = e.target.elements.email.value.trim();
    const password = e.target.elements.password.value.trim();
    loginUser(email, password);
  };
  const token = localStorage.hasOwnProperty('token')
    ? localStorage.getItem('token')
    : null;
  const role = token != null ? jwtDecode(token).userRole : '';
  const page = role === 'ADMIN' ? `/admin` : `/products`;
  return (
    <Fragment>
      <div>
        <div className="formbox">
          {loginState === constants.LOGGING_IN &&
            'please wait checking your credentials ...'}
          {loginState === constants.LOGIN_SUCCESS && (
            <div>
              Login was Successful. <Redirect to={page} />{' '}
            </div>
          )}
          <div className="log_head">
            <h1>Login Here</h1>
          </div>

          <form onSubmit={onFormSubmit}>
            <p>Email</p>
            <input
              type="email"
              name="email"
              placeholder="Enter  Email"
              id="email"
            />
            <p>Password</p>
            <input
              type="password"
              name="password"
              placeholder="Enter password"
              id="password"
            />
            <input type="submit" name="" value="submit" />
          </form>
          <div className="form_footer">
            <p>Store Manager copyright &copy 2018 </p>
          </div>
        </div>
      </div>
    </Fragment>
  );
};

export default LoginComponent;
