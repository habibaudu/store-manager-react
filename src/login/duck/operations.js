import axios from 'axios';
import { toast } from 'react-toastify';
import actions from './actions';
import constants from './constants';

const API_URL = process.env.API_URL;
const { setLoginState, setLoginError } = actions;
const url = `${API_URL}auth/login`;

const doLogin = (email, password) => dispatch => {
  dispatch(setLoginState(constants.LOGGING_IN));
  dispatch(setLoginError(''));
  return axios
    .post('https://store-manager2.herokuapp.com/api/v1/auth/login', {
      email,
      password,
    })
    .then(({ data }) => {
      localStorage.setItem('token', data.token);
      dispatch(setLoginState(constants.LOGIN_SUCCESS, data));
    })
    .catch(({ response }) => {
      dispatch(setLoginState(constants.LOGIN_ERROR));
      dispatch(setLoginError(response));
      toast.error(response, {
        hideProgressBar: true,
      });
    });
};

export default doLogin;
