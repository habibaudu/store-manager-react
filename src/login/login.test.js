import React from 'react';
import thunk from 'redux-thunk';
import axios from 'axios';
import configureStore from 'redux-mock-store';
import { Redirect, MemoryRouter } from 'react-router-dom';
import { shallow, mount } from 'enzyme';
import { Provider } from 'react-redux';
import { LoginComponent as Login } from './LoginComponent';
import { actions, constants, loginReducer, types } from './duck';
import { EllipsisLoaderComponent } from '../loaders';
import {
  LoginContainer,
  mapDispatchToProps,
  mapStateToProps,
} from './LoginContainer';

jest.mock('axios');

const { setLoginState, setLoginError } = actions;

describe('Login Component', () => {
  it('should render without throwing an error', () => {
    const loginUser = jest.fn();
    const props = {
      loginUser,
      loginState: '',
      errorMessage: 'Invalid credentials',
      history: {},
    };
    const wrapper = shallow(<Login {...props} />);
    wrapper.find('form').simulate('submit', {
      target: {
        elements: {
          email: {
            value: 'hello',
          },
          password: {
            value: 'pass',
          },
        },
      },
      preventDefault: () => {},
    });
    expect(loginUser).toHaveBeenCalledWith('hello', 'pass');
  });

  it('should render an email input', () => {
    expect(shallow(<Login />).find('#email').length).toEqual(1);
  });

  it('should render a password input', () => {
    expect(shallow(<Login />).find('#password').length).toEqual(1);
  });
});
