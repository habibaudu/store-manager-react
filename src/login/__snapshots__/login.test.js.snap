// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Component should redirect page if login is successful 1`] = `
<Fragment>
  <div>
    <div
      className="formbox"
    >
      <div>
        Login was Successful. 
        <Redirect
          push={false}
          to="/products"
        />
         
      </div>
      <div
        className="log_head"
      >
        <h1>
          Login Here
        </h1>
      </div>
      <form
        onSubmit={[Function]}
      >
        <p>
          Email
        </p>
        <input
          id="email"
          name="email"
          placeholder="Enter  Email"
          type="email"
        />
        <p>
          Password
        </p>
        <input
          id="password"
          name="password"
          placeholder="Enter password"
          type="password"
        />
        <input
          name=""
          type="submit"
          value="submit"
        />
      </form>
      <div
        className="form_footer"
      >
        <p>
          Store Manager copyright &copy 2018 
        </p>
      </div>
    </div>
  </div>
</Fragment>
`;
